pmd-java/src/main/resources/category/java/design.xml

```diff
@@ -1727,6 +1727,38 @@ public class MaybeAUtility {
   public static void foo() {}
   public static void bar() {}
 }
+]]>
+        </example>
+    </rule>
+
+    <rule name="MutableStaticState"
+          language="java"
+          since="6.35.0"
+          message="Do not use non-final non-private static fields"
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_design.html#mutablestaticstate">
+        <description>
+Non-private static fields must be made constants (or immutable references) by
```

Comment:
We probably should add some reason, why the fields must be constants... actually, I would phrase it "... should be made constants...", more soft, but that's maybe just my preference - there might be valid exceptions, in which case you probably will need to suppress this rule on such special cases.

Maybe something like: "Such fields break encapsulation and can lead to hard to find bugs, since these fields can be modified from anywhere within the program."?